src/simplay/src/simplay/visualization.py

```diff
@@ -29,11 +35,85 @@ def set_area(
         :param y: The y coordinate (row) of the top left corner of the area.
         :param color: The color of the area.
         """
+        if not isinstance(id, str):
+            raise ValueError("id must be a string")
+        if not isinstance(name, str):
+            raise ValueError("name must be a string")
+        if not isinstance(height, int):
+            raise ValueError("height must be an integer")
+        if not isinstance(width, int):
+            raise ValueError("width must be an integer")
+        if not isinstance(x, int):
+            raise ValueError("x must be an integer")
+        if not isinstance(y, int):
+            raise ValueError("y must be an integer")
+        if not isinstance(color, int):
+            raise ValueError("color must be an integer")
+        if x < 0 or x >= self.cols:
```

Comment:
maybe it makes sense to put all the following expectations also in the doc, so users are aware of the limitations

Reply:
I think in this case this should be rather intuitive, but I agree with you that we could add this clarification.